api/types/app.go

```diff
@@ -400,6 +414,11 @@ func (a *AppV3) CheckAndSetDefaults() error {
 			return trace.BadParameter("app %q invalid label key: %q", a.GetName(), key)
 		}
 	}
+
+	if a.Spec.MCP != nil && a.Spec.MCP.Command != "" {
+		a.Spec.URI = SchemaMCPStdio
+	}
```

Comment:
To allow overriding URI if needed:

```suggestion
	if a.Spec.MCP != nil && a.Spec.MCP.Command != "" && a.Spec.URI == "" {
		a.Spec.URI = SchemaMCPStdio
	}
```